### PR TITLE
send shutdown event when dll is unloaded and EvaluationStop, SessionC…

### PR DESCRIPTION
**Pull request by zhangxiang1993**
image

### CODE REVIEW

File: onnxruntime/core/session/inference_session.cc [hunk 1]
```diff
@@ -1053,13 +1054,21 @@ Status InferenceSession::Run(const RunOptions& run_options, const std::vector<st
...
-
+  const Env& env = Env::Default();
   try {
     if (!is_inited_) {
       LOGS(*session_logger_, ERROR) << "Session was not initialized";
       return Status(common::ONNXRUNTIME, common::FAIL, "Session not initialized.");
     }
 
+    // check the frequency to send Evalutaion Stop event
+    if (TimeDiffMicroSeconds(time_sent_last_evalutation_start_) > kDurationBetweenSendingEvaluationStart) {
```
Comment (shuweihsu) [inline onnxruntime/core/session/inference_session.cc]: Does the time units match here?
The function name indicate the unit is ms, but it looks like the unit of kDurationBetweenSendingEvaluationStart is us.

Comment (shuweihsu) [reply onnxruntime/core/session/inference_session.cc]: It is matching.

Comment (zhangxiang1993) [reply onnxruntime/core/session/inference_session.cc]: I double checked it. It should be right. All is measured in microseconds(us):smile: 

---
In reply to: [360187176](https://github.com/microsoft/onnxruntime/pull/2704#discussion_r360187176) [](ancestors = 360187176)

File: onnxruntime/core/session/inference_session.h [hunk 2]
```diff
@@ -510,6 +510,10 @@ class InferenceSession {
   const long long kDurationBetweenSending = 1000 * 1000 * 60 * 10;  // duration in (us).  send a report every 10 mins
   std::string event_name_;                                          // where the model is loaded from: ["model_loading_uri", "model_loading_proto", "model_loading_istream"]
 
+  TimePoint time_sent_last_evalutation_start_;
+  const long long kDurationBetweenSendingEvaluationStart = 1000 * 50; // duration in (us). send a EvaluationStop Event every 50 ms;
```
Comment (martinb35) [inline onnxruntime/core/session/inference_session.h]: >duration in (us). send a EvaluationStop Event every 50 ms; [](start = 73, length = 58)

how did you decide on this number? can you add a comment saying why it's every 50 ms?

**Pull request merged**